src/ExtranetBundle/objects/db/TDataExtranetUser.class.php

```diff
@@ -598,9 +605,8 @@ public function DirectLogin($sLoginName, $sPassword, $callPostLoginHook = false)
         $this->AllowEditByAll(true);
         $bWasLoggedIn = $this->Login($sLoginName, $sPassword);
         $this->AllowEditByAll(false);
-        if ($bWasLoggedIn && $callPostLoginHook) {
-            $this->PostLoginHook();
-        }
+        
+        $this->setCallPostLoginHook($callPostLoginHook);
```

Comment:
At this point, setting setCallPostLoginHook is too late, as the postLoginHook is already called within the login function. 

As a result, the value is "true" by default and the PostLoginHook is always executed although we get from the parameter that it should be false?

Reply:
You are absolutely right. Fixed the position. Good catch! Thanks!